pilot/cmd/pilot-agent/main.go

```diff
@@ -145,15 +145,23 @@ var (
 			}
 
 			if len(proxyIP) != 0 {
-				role.IPAddresses = append(role.IPAddresses, proxyIP)
+				role.IPAddresses = []string{proxyIP}
 			} else if podIP != nil {
-				role.IPAddresses = append(role.IPAddresses, podIP.String())
+				role.IPAddresses = []string{podIP.String()}
 			}
 
 			// Obtain all the IPs from the node
 			if ipAddrs, ok := proxy.GetPrivateIPs(context.Background()); ok {
 				log.Infof("Obtained private IP %v", ipAddrs)
-				role.IPAddresses = append(role.IPAddresses, ipAddrs...)
+				if len(role.IPAddresses) == 1 {
```

Comment:
is deduping not helping this case?

Reply:
yes, the order matters. The podip must be the first one i think

Reply:
Ok. Makes sense. Then why don't need else block right? Any way by the time it comes here `len(role.IPAddresses)` will always be 1. Also can you please add comment about the importance of order here?

Reply:
I am not sure when it can be empty, you can see L168

Reply:
Yes. It can be empty or 1 - So We do not have to append incase it is empty right? Am I missing some thing?

Reply:
I see your point. Ignore the above comment